examples/satellites/check_stability.py

```diff
@@ -0,0 +1,188 @@
+              'bool', in_top_ten('DSP 20 (USA 149)')),
+             ("SDS III-6 in top 10 unlikely periods",
+              'bool', in_top_ten('SDS III-6')),
+             ("SDS III-7 in top 10 unlikely periods",
+              'bool', in_top_ten('SDS III-7')),
+             ("Orion 6 in top 10 unlikely periods",
+              'bool', in_top_ten('Advanced Orion 6')),
+             ("DSP 20 in top 5 unlikely periods",
+              'bool', in_top_five('DSP 20 (USA 149)')),
+             ("SDS III-6 in top 5 unlikely periods",
+              'bool', in_top_five('SDS III-6')),
+             ("SDS III-7 in top 5 unlikely periods",
+              'bool', in_top_five('SDS III-7')),
+             ("Orion 6 in top 5 unlikely periods",
+              'bool', in_top_five('Advanced Orion 6')),
+         ]
+
+def orbit_type_imputation_probes(bdb):
+    bdb.execute('''
+        CREATE TEMP TABLE inferred_orbit AS
+        INFER EXPLICIT
+        anticipated_lifetime, perigee_km, period_minutes, class_of_orbit,
+        PREDICT type_of_orbit
+            AS inferred_orbit_type
+            CONFIDENCE inferred_orbit_type_conf
+        FROM satellites_cc
+        WHERE type_of_orbit IS NULL;
+    ''')
+    query = '''
+        SELECT i.class_of_orbit, i.inferred_orbit_type,
+            COUNT(*) AS count, AVG(i.inferred_orbit_type_conf) AS mean,
+            AVG((i.inferred_orbit_type_conf - sub.answer) * (i.inferred_orbit_type_conf - sub.answer)) as variance
+        FROM inferred_orbit AS i,
+            (SELECT i.class_of_orbit, i.inferred_orbit_type,
+                AVG(i.inferred_orbit_type_conf) AS answer
+            FROM inferred_orbit as i
+            GROUP BY i.class_of_orbit, i.inferred_orbit_type) AS sub
+        WHERE i.class_of_orbit == sub.class_of_orbit
+            AND i.inferred_orbit_type == sub.inferred_orbit_type
+        GROUP BY i.class_of_orbit, i.inferred_orbit_type
+        ORDER BY i.class_of_orbit, i.inferred_orbit_type
+    '''
+    answers = dict(((cl, tp), (ct, avg, var))
+                   for (cl, tp, ct, avg, var)
+                   in bdb.execute(query).fetchall())
+    def probe_gen():
+        for cl in ["Elliptical", "LEO"]:
+            for tp in ["Deep Highly Eccentric",
+                       "Intermediate",
+                       "Molniya",
+                       "Sun-Synchronous",
+                       "Cislunar",
+                       "N/A"]:
+                (ct, avg, var) = answers.get((cl, tp), (0,0,0))
+                yield ("%s %s ct imputed instances" % (cl, tp),
+                       'num', float(ct))
+                yield ("%s %s mean inference confidence" % (cl, tp),
+                       'num', avg)
+                yield ("%s %s inference confidence stddev" % (cl, tp),
+                       'num', math.sqrt(var))
+    return list(probe_gen())
+
+######################################################################
+## Driver                                                           ##
```

Comment:
Not sure what you mean.  The files whose names I wish to parse are the intermediate analysis files, which do have metadata files associated with them.  However, the reason I am parsing their names is that analyze records the file names as part of the key in the results table, which is where plot gets them from; and if this is being distributed, the metadata files may not be available by that point.

The reason this whole thing is this messy is that I was ambivalent about encoding the convention that bdb files correspond to different levels of completed crosscat analysis.  On the one hand, probing doesn't care about that, and can just preserve the file names (which it does).  On the other hand, plotting really wants to show analysis counts rather than file names on the x axis, so I do need to go from file name to analysis count at some point.  Hence the disaster of doing the remapping inside the driver program.